elasticdl/python/common/k8s_client.py

```diff
@@ -81,16 +82,20 @@ def __init__(
             self.cluster = cluster_spec_module.cluster
 
     def _watch(self):
-        stream = watch.Watch().stream(
-            self.client.list_namespaced_pod,
-            self.namespace,
-            label_selector=ELASTICDL_JOB_KEY + "=" + self.job_name,
-        )
-        for event in stream:
-            try:
-                self._event_cb(event)
-            except Exception:
-                traceback.print_exc()
+        while True:
+            stream = watch.Watch().stream(
+                self.client.list_namespaced_pod,
+                self.namespace,
+                label_selector=ELASTICDL_JOB_KEY + "=" + self.job_name,
+            )
+            for event in stream:
+                try:
+                    self._event_cb(event)
+                except Exception:
```

Comment:
Should we only catch particular set of exceptions instead of `Exception` which is too broad.